package/repos/codeberg.py

```diff
@@ -0,0 +1,190 @@
+logger = logging.getLogger(__name__)
+
+
+REPOSITORY_URL = "https://codeberg.org/api/v1/repos/{repository}"
+COMMITS_URL = "https://codeberg.org/api/v1/repos/{repository}/commits"
+COLLABORATORS_URL = "https://codeberg.org/api/v1/repos/{repository}/collaborators"
+
+
+@dataclass
+class ForgejoMetadata:
+    archived: bool
+    archived_at: str
+    description: str
+    forks_count: int
+    watchers_count: int
+
+
+@dataclass
+class ForgejoCommit:
+    sha: str
+    created: str
+
+
+class CodebergHandler(BaseHandler):
+    title = "Codeberg"
+    url_regex = "(https|git)://codeberg.org/"
+    url = "https://codeberg.org/"
+    repo_regex = r"(?:https|git)://codeberg.org/[^/]*/([^/]*)"
+    slug_regex = repo_regex
+
+    def __init__(self):
+        self.api_token = settings.CODEBERG_TOKEN
+
+    def fetch_commits(self, package: "package.models.Package"):
+        from package.models import Commit  # Added here to avoid circular imports
+
+        for commit in _fetch_commits(package.repo_name()):
+            try:
+                _, created = Commit.objects.get_or_create(
+                    package=package,
+                    commit_hash=commit.sha,
+                    commit_date=commit.created,
+                )
+
+                # If the commit record already exists, it means we are at the end of the
+                # list we want to import
+                # TODO: same exit condition as github, fails for allauth
+                if not created:
+                    break
+            except Commit.MultipleObjectsReturned:
+                continue
+
+        package.save()
+
+        return package
+
+    def fetch_metadata(self, package: "package.models.Package"):
+        repo = _fetch_repository(package.repo_name())
+        collaborators = []
+
+        for user in _fetch_collaborators(package.repo_name(), self.api_token):
+            collaborators.append(user)
+
+        if repo.archived:
+            package.data_repo_archived = repo.archived_at
+
+        package.repo_description = repo.description
+        package.repo_forks = repo.forks_count
+        package.repo_watchers = repo.watchers_count
+        package.participants = ",".join(collaborators)
+        package.save()
+
+        return package
+
+
+def _fetch_repository(repository: str) -> ForgejoMetadata | None:
+    try:
+        url = REPOSITORY_URL.format(repository=repository)
+        response = httpx.get(url)
+    except Exception as exp:
+        logger.error(exp)
+        return None
+
+    if response.status_code != httpx.codes.OK:
+        logger.error(f"Response code: {response.status_code} for URL {url}")
+        return None
+
+    data = response.json()
+
+    try:
+        meta = ForgejoMetadata(
+            archived=data["archived"],
+            archived_at=data["archived_at"],
+            description=data["description"],
+            forks_count=data["forks_count"],
+            watchers_count=data["watchers_count"],  # TODO: watchers or starts?
```

Comment:
This field is re-written to `Stars` in Django admin. Should this data be stars, watchers or stars + watchers?

Reply:
The original version was written with GitHub API's terminology, so let's keep the watchers (publicly called stars). (we might refactor this later)